components/tools/OmeroPy/src/omero/gateway/__init__.py

```diff
@@ -141,6 +141,53 @@ def fileread_gen(fin, fsize, bufsize):
     fin.close()
 
 
+def countAnnotations(annotationlinks=None):
+    """
+    Count the different (unique) annotions from the
+    provided annotation links
+    """
+
+    counts = {
+        "TagAnnotation": 0,
+        "FileAnnotation": 0,
+        "CommentAnnotation": 0,
+        "LongAnnotation": 0,
+        "MapAnnotation": 0,
+        "OtherAnnotation": 0}
+
+    if annotationlinks is None:
+        return counts
+
+    atypes = {
+        omero.model.TagAnnotationI: "TagAnnotation",
+        omero.model.FileAnnotationI: "FileAnnotation",
+        omero.model.CommentAnnotationI: "CommentAnnotation",
+        omero.model.LongAnnotationI: "LongAnnotation",
+        omero.model.MapAnnotationI: "MapAnnotation"}
+
+    # Avoid counting the same annotations multiple times
+    uniqueIds = set([al._child._id._val for al in annotationlinks])
+
+    total = 0
+    regAnnotations = 0
+    for al in annotationlinks:
+        if al._child._id._val not in uniqueIds:
+            continue
+
+        total += 1
+        if type(al._child) in atypes:
```

Comment:
How shall I use `isinstance` to check if the type of the annotation is part of the dictionary `atypes`?